File: apps/mobile/test/settlement_progress_test.dart
import 'package:domain/domain.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:salda_mobile/features/sessions/domain/session_models.dart';

void main() {
  group('progreso de liquidación', () {
    test('completo usa las obligaciones necesarias y llega al 100 %', () {
      const progress = SettlementProgress(
        required: Money(6000),
        confirmed: Money(6000),
        marked: Money.zero,
      );

      expect(progress.confirmedFraction, 1);
      expect(progress.remaining, Money.zero);
      expect(progress.isSettled, isTrue);
    });

    test('parcial separa confirmado, marcado y pendiente', () {
      const progress = SettlementProgress(
        required: Money(6000),
        confirmed: Money(2000),
        marked: Money(1000),
      );

      expect(progress.confirmedFraction, closeTo(1 / 3, 0.0001));
      expect(progress.markedFraction, closeTo(1 / 6, 0.0001));
      expect(progress.remaining, const Money(4000));
      expect(progress.isSettled, isFalse);
    });

    test('sin transferencias necesarias define 0/0 como saldado', () {
      const progress = SettlementProgress(
        required: Money.zero,
        confirmed: Money.zero,
        marked: Money.zero,
      );

      expect(progress.confirmedFraction, 1);
      expect(progress.markedFraction, 0);
      expect(progress.remaining, Money.zero);
      expect(progress.isSettled, isTrue);
    });
  });
}

File: apps/guest_web/src/lib/settlement-progress.test.ts
import { describe, expect, it } from 'vitest';

import { settlementProgress } from './settlement-progress';

describe('progreso de liquidación', () => {
  it('usa obligaciones y no el gasto total', () => {
    const progress = settlementProgress({
      settlementRequired: 6000,
      settledConfirmed: 6000,
      settledMarked: 0,
    });
    expect(progress.confirmedFraction).toBe(1);
    expect(progress.isSettled).toBe(true);
  });

  it('separa lo marcado de lo confirmado', () => {
    const progress = settlementProgress({
      settlementRequired: 6000,
      settledConfirmed: 2000,
      settledMarked: 1000,
    });
    expect(progress.confirmedFraction).toBeCloseTo(1 / 3);
    expect(progress.markedFraction).toBeCloseTo(1 / 6);
    expect(progress.remaining).toBe(4000);
    expect(progress.isSettled).toBe(false);
  });

  it('0/0 significa saldado sin dividir por cero', () => {
    const progress = settlementProgress({
      settlementRequired: 0,
      settledConfirmed: 0,
      settledMarked: 0,
    });
    expect(progress.confirmedFraction).toBe(1);
    expect(progress.markedFraction).toBe(0);
    expect(progress.isSettled).toBe(true);
  });
});
